fiftyone/utils/youtube.py

```diff
@@ -0,0 +1,461 @@
+"""
+Utilities for working with `YouTube <https://youtube.com>`.
+
+| Copyright 2017-2022, Voxel51, Inc.
+| `voxel51.com <https://voxel51.com/>`_
+|
+"""
+import itertools
+import multiprocessing
+import multiprocessing.dummy
+import os
+
+import numpy as np
+
+import eta.core.utils as etau
+import eta.core.video as etav
+
+import fiftyone.core.utils as fou
+
+pytube = fou.lazy_import(
+    "pytube", callback=lambda: fou.ensure_package("pytube"),
+)
+
+
+logger = multiprocessing.get_logger()
+
+
+def download_youtube_videos(
+    urls,
+    download_dir=None,
+    video_paths=None,
+    clip_segments=None,
+    ext=".mp4",
```

Comment:
I think `ext` makes more sense as a single *hint* rather than a list of strict requirements, because in general the user would have no way to know what codecs might be available for a particular watch URL

Reply:
FYI- an `ext` argument is used quite a bit throughout the codebase, and in those places it always includes the `.`. The original reasoning was for consistency with what you get  `os.path.splitext()`, but at this point we should continue with the convention regardless just for API consistency.